src/main/java/control/ProjectController.java

```diff
@@ -72,6 +71,51 @@ public void saveAs() {
     }
     
     /**
+     * Handler for when the save button is clicked.
+     * @param event the MouseEvent for this handler
+     */
+    private void save(MouseEvent event) {
+        if (validateProjectData()) {
+            editProjectModal.hideModal();
+            
+            String name = editProjectModal.getNameField().getText();
+            String description = editProjectModal.getDescriptionField().getText();
+            String directorTimelineDescription = editProjectModal
+                    .getDirectorTimelineDescriptionField().getText();
+            double secondsPerCount = Double.parseDouble(
+                    editProjectModal.getSecondsPerCountField().getText());
+            
+            ScriptingProject project = new ScriptingProject(name, description, secondsPerCount);
+            project.setDirectorTimeline(new DirectorTimeline(directorTimelineDescription, null));
+            project.setCameraTypes(editProjectModal.getCameraTypes());
+            project.setCameras(editProjectModal.getCameras());
+            project.setCameraTimelines(editProjectModal.getTimelines());
+            project.getDirectorTimeline().setProject(project);
+            project.setFilePath(editProjectModal.getProject().getFilePath());
+            for (CameraTimeline timeline : project.getCameraTimelines()) {
+                timeline.setProject(project);
+            }
+            controllerManager.setScriptingProject(project);
+            controllerManager.updateWindowTitle();
+            RootCenterArea area = new RootCenterArea(
+                    controllerManager.getRootPane(), editProjectModal.getTimelines().size(), false);
+            controllerManager.getRootPane().reInitRootCenterArea(area);
+            for (int i = 0;i < project.getCameraTimelines().size();i++) {
+                CameraTimeline newLine = project.getCameraTimelines().get(i);
+                CameraTimeline oldLine = editProjectModal.getProject().getCameraTimelines().get(i);
+                LinkedList<CameraShot> shots = new LinkedList<CameraShot>();
+                for (CameraShot shot: oldLine.getShots()) {
```

Comment:
This can be replaced by a `.forEach()` call on `oldLine.getShots()` with a lambda.